src/out-resolve.js

```diff
@@ -1,12 +1,13 @@
 import path from 'path';
 
-export default (input, output) => new Promise(resolve => {
+export default (input, output, root, allInOutput) => new Promise(resolve => {
   if (output && path.extname(output)) {
     return resolve(output);
   }
 
   if (output) {
-    return resolve(path.join(output, path.basename(input)));
+    const inputPath = allInOutput ? path.relative(root, input) : path.basename(input);
```

Comment:
I don't really like ternaries because if there is more logic it will have to be expanded and it may become unreadable. We can replace this with

```diff
-    const inputPath = allInOutput ? path.relative(root, input) : path.basename(input);
+    let inputPath = path.basename(input);
+
+    if (allInOutput) {
+        inputPath = path.relative(root, input);
+    }
```